Userland/Applications/PixelPaint/MainWidget.cpp

```diff
@@ -121,6 +117,7 @@ void MainWidget::image_editor_did_update_undo_stack()
         m_redo_action->set_enabled(false);
         return;
     }
+    image_editor->update_modified();
```

Comment:
this and line 86 feel like bug fixes rather than part of the refactoring you're doing with the modification state. I wonder if they should be made in a separate" make sure to update modification state in X case" bugfix commit? up to you though.